File: mobile/src/main/java/treehou/se/habit/ui/servers/serverlist/ServersModule.java
package treehou.se.habit.ui.servers.serverlist;


import dagger.Module;
import dagger.Provides;
import treehou.se.habit.module.ViewModule;

@Module
public class ServersModule extends ViewModule<ServersFragment> {

    public ServersModule(ServersFragment fragment) {
        super(fragment);
    }

    @Provides
    public ServersContract.View provideView() {
        return view;
    }

    @Provides
    public ServersContract.Presenter providePresenter(ServersPresenter presenter) {
        return presenter;
    }
}

File: mobile/src/main/java/treehou/se/habit/ui/servers/serverlist/ServersModule.kt
package treehou.se.habit.ui.servers.serverlist


import dagger.Module
import dagger.Provides
import treehou.se.habit.module.ViewModule

@Module
class ServersModule(fragment: ServersFragment) : ViewModule<ServersFragment>(fragment) {

    @Provides
    fun provideView(): ServersContract.View {
        return view
    }

    @Provides
    fun providePresenter(presenter: ServersPresenter): ServersContract.Presenter {
        return presenter
    }
}
